docs/ape17.rst

```diff
@@ -0,0 +1,533 @@
+
+Next up, add a new section to the ``setup.cfg`` file (or modify, if it
+already exists), to specify optional dependencies as well as
+dependencies required to run tests and build the documentation, for
+example::
+
+   [options.extras_require]
+   all =
+       scipy
+       matplotlib
+   test =
+       pytest-astropy
+   docs =
+       sphinx-astropy
+
+If you don’t need any optional dependencies, remove the ``all`` section.
+You will likely need to have at least ``pytest-astropy`` in the ``test``
+section and ``sphinx-astropy`` in ``docs``.
+
+Step 4 - Define package data
+----------------------------
+
+If your package includes non-Python data files, you will need to update
+how you declare which data files to include. If you have been using the
+Astropy package template, it is likely that you have functions called
+``get_package_data`` defined inside ``setup_package.py`` files. Remove
+these functions, and instead define the package data using a
+``[options.package_data]`` section inside your ``setup.cfg`` file,
+e.g.::
+
+   [options.package_data]
+   * = *.fits, *.csv
+   my_package.tests = data/*
+
+In the above example, all ``.fits`` and ``.csv`` in the package will be
+included as well as all files inside ``my_package/tests/data``.
+
+Step 5 - Update your ``setup.py`` file
+--------------------------------------
+
+Copy the ``setup.py`` file you generated in :ref:`Step 0 <step0>` and replace your existing one
+- it should be good to go as-is without any further customizations.
+
+Step 6: add a ``pyproject.toml`` file
+-------------------------------------
+
+The ``pyproject.toml`` file is used to declare dependencies needed to run
+``setup.py`` and build the package. Copy the ``pyproject.toml`` file you
+generated in :ref:`Step 0 <step0>` and replace your existing one.
+
+If your package doesn’t have any compiled extensions, the file should contain:
+
+.. code:: toml
+
+   [build-system]
+   requires = ["setuptools",
+               "setuptools_scm",
+               "wheel"]
+   build-backend = 'setuptools.build_meta'
+
+Step 7 - Handling C/Cython extensions
+-------------------------------------
+
+If your package has no compiled C/Cython extensions, you can skip this
+step. Otherwise, if you have C or Cython extensions, you can either
+define your extensions manually inside the ``setup.py`` file or make use
+of the `extension-helpers <https://extension-helpers.readthedocs.io>`__
+package to collect extensions in a similar way to astropy-helpers.
+
+Step 7a - Defining extensions manually
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can define extensions manually as described
+`here <https://oa-packaging-guide-preview.readthedocs.io/en/latest/extensions.html#defining-extensions-in-setup-py>`__.
+If you do this, you can remove all ``setup_package.py`` files in your
+package, and you don't need to include extension-helpers in the
+``pyproject.toml`` file.
+
+If you have Cython extensions or your extensions use the NumPy C API,
```

Comment:
I think this sentence is duplicated in 7b and it makes more sense living in 7b.
```suggestion
```

Reply:
I'd like to keep this, because people will read *either* 7a or 7b (but won't do both) so it's important to mention this in both cases.